### PR TITLE
Add tests for `ProxyFactoryDecoder`

### PR DESCRIPTION
**Pull request by iamacook**
This adds tests for the `decodeFunctionData` method of the `ProxyFactoryDecoder`.

### CODE REVIEW

File: src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { Hex } from 'viem';
+import { faker } from '@faker-js/faker';
+import { ProxyFactoryDecoder } from '@/domain/relay/contracts/proxy-factory-decoder.helper';
+import { createProxyWithNonceEncoder } from '@/domain/relay/contracts/__tests__/proxy-factory-encoder.builder';
+
+describe('ProxyFactoryDecoder', () => {
+  let target: ProxyFactoryDecoder;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
```
Comment (fmrsabino) [inline src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts:10]: To be consistent with the recent change [5b0e668](https://github.com/safe-global/safe-client-gateway/commit/5b0e668196b433b1fae7297180ddd52807f821c8)
```suggestion
    jest.resetAllMocks();
```

File: src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    target = new ProxyFactoryDecoder();
+  });
+
+  it('decodes a createProxyWithNonce function call correctly', () => {
+    const createProxyWithNonce = createProxyWithNonceEncoder();
+    const args = createProxyWithNonce.build();
+    const data = createProxyWithNonce.encode();
```
Comment (fmrsabino) [inline src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts:17]: nitpick:
```
// arrange

// act

// assert
```

File: src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  let target: ProxyFactoryDecoder;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    target = new ProxyFactoryDecoder();
+  });
+
+  it('decodes a createProxyWithNonce function call correctly', () => {
+    const createProxyWithNonce = createProxyWithNonceEncoder();
+    const args = createProxyWithNonce.build();
```
Comment (fmrsabino) [inline src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts:16]: I would move this to the "assert" section and call it `expectedArgs` (as it's only used in the assertion). What do you think?

**Review by fmrsabino — CHANGES_REQUESTED** [review]

File: src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts (hunk 1 above)
Comment (iamacook) [reply src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts]: Changed in 192ff518bad0aff21b70ad12ca180db3b701020b.

File: src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts (hunk 2 above)
Comment (iamacook) [reply src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts:16]: I think I restructured it correctly in 192ff518bad0aff21b70ad12ca180db3b701020b.

File: src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts (hunk 3 above)
Comment (iamacook) [reply src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts]: Makes sense to me - changed in 192ff518bad0aff21b70ad12ca180db3b701020b.

File: src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts [hunk 4]
```diff
@@ -7,18 +7,23 @@ describe('ProxyFactoryDecoder', () => {
...
     target = new ProxyFactoryDecoder();
   });
 
   it('decodes a createProxyWithNonce function call correctly', () => {
     const createProxyWithNonce = createProxyWithNonceEncoder();
-    const args = createProxyWithNonce.build();
     const data = createProxyWithNonce.encode();
 
+    const expectedArgs = createProxyWithNonce.build();
+
```
Comment (fmrsabino) [inline src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts:19]: nitpick:
```suggestion
```

**Review by fmrsabino — APPROVED** [review]
🚀

File: src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts (hunk 4 above)
Comment (iamacook) [reply src/domain/relay/contracts/__tests__/proxy-factory-decoder.helper.spec.ts]: Removed in 2b75f622363761d6856fe5e53d70bcc1d0988005.

**Review by fmrsabino — APPROVED** [review]
👏 🚀

**Pull request merged**